src/Compilers/Core/Portable/Serialization/StreamObjectReader.cs

```diff
@@ -93,6 +101,30 @@ internal sealed partial class StreamObjectReader : ObjectReader, IDisposable
             }
         }
 
+        /// <summary>
+        /// Attempts to create a <see cref="StreamObjectReader"/> from the provided <paramref name="stream"/>.
+        /// If the <paramref name="stream"/> does not start with a valid header, then <code>null</code> will
+        /// be returned.
+        /// </summary>
+        public static StreamObjectReader TryGetReader(
+            Stream stream,
+            ObjectData knownObjects = null,
+            ObjectBinder binder = null,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (stream == null)
+            {
+                return null;
+            }
+
+            if (stream.ReadByte() != VersionByte1 && stream.ReadByte() != VersionByte2)
+            {
+                return null;
+            }
+
+            return new StreamObjectReader(stream, knownObjects, binder, cancellationToken);
```

Comment:
I could.  But that indicates a true bug.  That means the format changed *and* the version wasn't updated.  That's actually a bug, and i want an exceptoin in that case.